ecs-cli/modules/cli/compose/project/project_parseV3.go

```diff
@@ -11,20 +17,93 @@ import (
 func (p *ecsProject) parseV3() (*[]containerconfig.ContainerConfig, error) {
 	logrus.Debug("Parsing v3 project...")
 
-	// TODO: parse v3, convert ServiceConfigs to ContainerConfigs
-	configDetails := types.ConfigDetails{}
+	// load and parse each compose file into a configFile
+	configFiles := []types.ConfigFile{}
+	for _, file := range p.ecsContext.ComposeFiles {
+
+		loadedFile, err := ioutil.ReadFile(file)
+		if err != nil {
+			return nil, err
+		}
+		parsedFile, err := loader.ParseYAML(loadedFile)
+		if err != nil {
+			return nil, err
+		}
+		configFile := types.ConfigFile{
+			Filename: file,
+			Config:   parsedFile,
+		}
+		configFiles = append(configFiles, configFile)
+	}
+
+	// get build environment
+	env, err := getLocalEnv()
+	if err != nil {
+		return nil, err
+	}
+
+	// get compose working dir
+	wrkDir, err := getWorkingDir(p.ecsContext.ComposeFiles[0])
+	if err != nil {
+		return nil, err
+	}
+
+	// build config details
+	configDetails := types.ConfigDetails{
+		WorkingDir:  wrkDir,
+		ConfigFiles: configFiles,
+		Environment: env,
```

Comment:
for some reason I'm getting the heebie-jeebies about this. I realize that this is what docker/cli defines on `ConfigDetails` but unless i'm missing something it seems a little weird that we're copying the customer's whole environment into here?

Reply:
it looks like is the Env is not supplied, Docker makes sys call to get the LookupEnv:
https://github.com/docker/cli/blob/master/cli/compose/interpolation/interpolation.go#L31
 
Given this, I'm inclined to leave the sys calling to Docker unless we find a compelling reason to set it ourselves.